test/serializers/serialization_array_test.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module ActiveModel
+  class Serializer
+    class SerializationArrayTest < ActiveSupport::TestCase
+      class ArraySerializer < ActiveModel::Serializer
+        attributes :items
+
+        def items
+          CollectionSerializer.new(object).as_json
+        end
+      end
+
+      class Author < ActiveModelSerializers::Model
+        attributes :id, :name
+      end
+      class AuthorSerializer < ActiveModel::Serializer
+        attributes :id, :name
+      end
+
+      setup do
+        @authors = [Author.new(id: 1, name: 'Blog Author')]
+        @serializer_instance = ArraySerializer.new(@authors)
+        @serializable = ActiveModelSerializers::SerializableResource.new(@authors, serializer: ArraySerializer, adapter: :attributes)
```

Comment:
if we change the `adapter` to `:json` and pass an option like `root: 'items'` to meet https://github.com/rails-api/active_model_serializers/pull/2329/files#diff-d81a4fb04ad7026e9924afee4099ddbfR6, then **the tests will pass without any change in the lib/active_model/serializer.rb**
```ruby
@serializable = ActiveModelSerializers::SerializableResource.new(
  @authors,
  adapter: :json,
  root: 'items'
)
```

If the serializer class cannot be determined for the instances of the array `@authors`, e.g. the name of the serializer is `AuthorPreviewSerializer` instead of `AuthorSerializer`, then we need to pass the name using `each_serializer` option like this way:
```ruby
@serializable = ActiveModelSerializers::SerializableResource.new(
  @authors,
  each_serializer: AuthorPreviewSerializer,
  adapter: :json,
  root: 'items'
)
```
More information on the `root` key can be found here: https://github.com/rails-api/active_model_serializers/blob/0-10-stable/docs/general/rendering.md#overriding-the-root-key